infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -141,9 +140,20 @@
     private static Counter counterTokenSum;
     private static Counter counterToken0;
 
+    @Getter
+    @Setter
+    private volatile long readyStateEpoch = -1;
+
     @Override
-    public boolean isServerReady() {
-        return serverContext.isReady();
+    public boolean isServerReadyToHandleMsg(CorfuMsg msg) {
+        if (readyStateEpoch != serverContext.getServerEpoch()) {
+            if (!msg.getMsgType().equals(CorfuMsgType.BOOTSTRAP_SEQUENCER)) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1066)